nncf/torch/quantization/quantize_functions.py

```diff
@@ -126,23 +126,19 @@ def forward(ctx, input_, input_shape, scale, level_low, level_high, levels):
         input_low = torch.where(scale > 0, -scale, -scale / level_low * level_high)
         # 15/8 * scale or (2-1/8) * scale
         input_range = torch.abs((2 + 1 / level_low) * scale)
-
-        if input_.dtype in [torch.bfloat16, torch.float16]:
-            input_low = input_low.type(input_.dtype)
-            input_range = input_range.type(input_.dtype)
-
+        dtype = input_.dtype
         original_shape = input_.shape
         input_ = input_.reshape(input_shape)
 
-        output = RQ.Quantize_forward(input_, input_low, input_range, levels)
+        output = RQ.Quantize_forward(input_.type(torch.float32), input_low, input_range, levels)
```

Comment:
Why not pass all arguments in float32 precision?

Reply:
input_low, input_range is already in float32